Twister broken with external tree that specifies soc_root in zephyr/module.yml

The [`zephyr-silabs`][1] repository [use zephyr as an external tree][2]. `zephyr-silabs` recently [update from Zephyr commit a2ac676a1 to 9f93dede3][3]. Since then, [`zephyr-silabs` CI is broken][4] (I also wonder why twister fails but the job successes, but it is out of the scope of this bug report).

I am able to reproduce the issue by running the same commands than the [`silabs-zephyr` CI][5]. To summary:

    west init -m git@github.com:siliconlabssoftware/zephyr-silabs
    west update
    west twister -s sample.bluetooth.peripheral_hr -p xg27_dk2602a

Using git bisect, I have found the behavior of twister has changed with the commit f1bf24e4d ("boards: other: stm32_min_dev: use one twister.yaml"). I am also able to fix the issue by reverting these commits:

    "boards: other: stm32_min_dev: use one twister.yaml"
    "twister: rework board handling"
    "boards: intel_adsp: use one twister.yaml"
    "twister: test udpates for new board handling"
    "twister: --device-testing always implies --filter runnable"
    "twister: fix platform filter when loading plan from file"
    "twister: also convert platform names from the hardwaremap"

[1]: https://github.com/SiliconLabsSoftware/zephyr-silabs/
[2]: https://github.com/SiliconLabsSoftware/zephyr-silabs/blob/main/west.yml
[3]: https://github.com/SiliconLabsSoftware/zephyr-silabs/commit/d87b287c057d1184cc9fea1235cd5264e6234e22#diff-d7f4082837dc3cb270fcf1abd68188216fdadee294e01611b8ca59ddea7fdbb3
[4]: https://github.com/SiliconLabsSoftware/zephyr-silabs/actions/runs/11547973881/job/32138664649
[5]: https://github.com/SiliconLabsSoftware/zephyr-silabs/actions/runs/11547973881/workflow

Pasting here the relevant part of the CI failure logs:
```
Run west twister -s sample.bluetooth.peripheral_hr -p siwx917_rb4338a -v --inline-logs -K
Renaming output directory to /home/runner/work/zephyr-silabs/zephyr-silabs/zephyr-silabs/twister-out.1
INFO    - Using Ninja..
INFO    - Zephyr version: 9f93dede369c
INFO    - Using 'zephyr' toolchain.
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.10/x64/bin/west", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/west/app/main.py", line 1085, in main
    app.run(argv or sys.argv[1:])
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/west/app/main.py", line 244, in run
    self.run_command(argv, early_args)
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/west/app/main.py", line 505, in run_command
    self.run_extension(args.command, argv)
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/west/app/main.py", line 654, in run_extension
    self.cmd.run(args, unknown, self.topdir, manifest=self.manifest,
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/site-packages/west/commands.py", line 194, in run
    self.do_run(args, unknown)
  File "/home/runner/work/zephyr-silabs/zephyr-silabs/zephyr/scripts/west_commands/twister_cmd.py", line 61, in do_run
    ret = main(options, default_options)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/zephyr-silabs/zephyr-silabs/zephyr/scripts/pylib/twister/twisterlib/twister_main.py", line 120, in main
    tplan.discover()
  File "/home/runner/work/zephyr-silabs/zephyr-silabs/zephyr/scripts/pylib/twister/twisterlib/testplan.py", line 185, in discover
    self.add_configurations()
  File "/home/runner/work/zephyr-silabs/zephyr-silabs/zephyr/scripts/pylib/twister/twisterlib/testplan.py", line 424, in add_configurations
    known_boards = list_boards.find_v2_boards(lb_args)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/zephyr-silabs/zephyr-silabs/zephyr/scripts/list_boards.py", line 329, in find_v2_boards
    systems = list_hardware.find_v2_systems(root_args)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/zephyr-silabs/zephyr-silabs/zephyr/scripts/list_hardware.py", line 238, in find_v2_systems
    for root in unique_paths(args.soc_roots):
  File "/home/runner/work/zephyr-silabs/zephyr-silabs/zephyr/scripts/list_hardware.py", line 204, in unique_paths
    yield from dict.fromkeys(map(Path.resolve, paths)).keys()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.10/x64/lib/python3.11/pathlib.py", line 997, in resolve
    p = self._from_parts((s,))
        ^^^^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute '_from_parts'
Error: Process completed with exit code 1.
```

--- a/scripts/pylib/twister/twisterlib/environment.py
+++ b/scripts/pylib/twister/twisterlib/environment.py
@@ -980,13 +980,13 @@ class TwisterEnv:
         for module in modules:
             soc_root = module.meta.get("build", {}).get("settings", {}).get("soc_root")
             if soc_root:
-                self.soc_roots.append(os.path.join(module.project, soc_root))
+                self.soc_roots.append(Path(module.project) / Path(soc_root))
             dts_root = module.meta.get("build", {}).get("settings", {}).get("dts_root")
             if dts_root:
-                self.dts_roots.append(os.path.join(module.project, dts_root))
+                self.dts_roots.append(Path(module.project) / Path(dts_root))
             arch_root = module.meta.get("build", {}).get("settings", {}).get("arch_root")
             if arch_root:
-                self.arch_roots.append(os.path.join(module.project, arch_root))
+                self.arch_roots.append(Path(module.project) / Path(arch_root))
 
         self.hwm = None
 

